### PR TITLE
[ESP32]Add CI for ESP32C3

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
CI need to be add since C3 support has be added for all-clusters-app

#### Change overview
Add CI for ESP32C3-DevKitM

#### Testing
Successful CI

### CODE REVIEW

**Comment by wqx6** [comment]
@andy31415 Can you please updated the docker hub? we have update ESP-IDF version to v4.3 in #6345 .

**Review by woody-apple — APPROVED** [review]

**Comment by woody-apple** [comment]
@saurabhst @msandstedt @Damian-Nordic @jepenven-silabs @mspang @andy31415 ?

File: examples/all-clusters-app/esp32/sdkconfig_c3devkit.defaults [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+#enable lwip ipv6 autoconfig
+CONFIG_LWIP_IPV6_AUTOCONFIG=y
+
+# Use a custom partition table
+CONFIG_PARTITION_TABLE_CUSTOM=y
+CONFIG_PARTITION_TABLE_FILENAME="partitions.csv"
+
+# Main task needs a bit more stack than the default
+# default is 3584, bump this up to 4k.
+CONFIG_ESP_MAIN_TASK_STACK_SIZE=4096
```
Comment (dhrishi) [inline examples/all-clusters-app/esp32/sdkconfig_c3devkit.defaults:45]: I think this option won't be required for C3. It was required for enabling on IP network commissioning using the M5Stack

File: scripts/examples/esp_example.sh [hunk 2]
```diff
@@ -36,14 +36,17 @@ source "$IDF_PATH/export.sh"
...
+
     rm -f "$root"/sdkconfig
     (
         cd "$root"
+        idf.py set-target "$idf_target"
         idf.py -D SDKCONFIG_DEFAULTS="$sdkconfig_name" build
-    )
-    (
-        cd "$root"
-        idf.py build "$@"
```
Comment (dhrishi) [inline scripts/examples/esp_example.sh:46]: This is removed as a part of the PR #7910 .

**Review by dhrishi — CHANGES_REQUESTED** [review]

File: examples/all-clusters-app/esp32/sdkconfig_c3devkit.defaults (hunk 1 above)
Comment (wqx6) [reply examples/all-clusters-app/esp32/sdkconfig_c3devkit.defaults]: Have deleted it.

File: scripts/examples/esp_example.sh (hunk 2 above)
Comment (wqx6) [reply scripts/examples/esp_example.sh]: Rebased

**Review by dhrishi — APPROVED** [review]

**Review by msandstedt — APPROVED** [review]

**Review by mspang — APPROVED** [review]

**Pull request merged**